### PR TITLE
Prime secrets list cache on startup

### PR DESCRIPTION
**Pull request by csstaub**
@mcpherrinm

### CODE REVIEW

**Comment by mcpherrinm** [comment]
lgtm

**Comment by mcpherrinm** [comment]
You should obey the setting
 	ping           = flag.Bool("ping", false, "Enable startup ping to server")

**Comment by csstaub** [comment]
@mcpherrinm done

**Comment by mcpherrinm** [comment]
hmm, ping will now do two requests to the server for secrets list at startup.  Seems like we should combine the logic for both.

**Comment by csstaub** [comment]
Hm, yeah. We should drop the ping in new client. There's no access to the cache there. If we're gonna talk to the server we should cache the result, too.

**Comment by mcpherrinm** [comment]
:+1: to removing the client ping

**Comment by csstaub** [comment]
@mcpherrinm PTAL

File: cache.go [hunk 1]
```diff
@@ -252,3 +252,23 @@ func (c *Cache) backendSecretList() chan []Secret {
...
+
+func (c *Cache) pingBackend() bool {
+	secrets, ok := c.backend.SecretList()
+	if !ok {
+		return false
+	}
+
+	newMap := NewSecretMap()
+	for _, backendSecret := range secrets {
+		// If the cache contains a secret with content, keep it over backendSecret.
```
Comment (mcpherrinm) [inline cache.go]: I'm not clear why we'd prefer the cache over the freshly fetched data.

**Comment by mcpherrinm** [comment]
lgtm

**Pull request merged**